bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -198,14 +239,38 @@ public Collection<ProductListItem> getProducts() {
         if (usePlaceholderData) {
             CategoryInterface category = getCategory();
             CategoryProducts categoryProducts = category.getProducts();
-            ProductToProductListItemConverter converter = new ProductToProductListItemConverter(productPage, request, urlProvider, getId(),
+            ProductToProductListItemConverter converter = new ProductToProductListItemConverter(productPage, request,
+                urlProvider, getId(),
                 category);
             return categoryProducts.getItems().stream()
                 .map(converter)
                 .filter(Objects::nonNull) // the converter returns null if the conversion fails
                 .collect(Collectors.toList());
         } else {
-            return getSearchResultsSet().getProductListItems();
+            Collection<ProductListItem> products = getSearchResultsSet().getProductListItems();
```

Comment:
Iiuc this adds all the fragments to every page, right? We should check with Martin if this was then intention. I understood that I I place a fragment at position=3 (before the 3rd product) it will be shown on the first page in the 3rd position but not on the second page.

Reply:
Yes, I asked Martin to confirm. 

It depends. We have some options there, if needed we can discuss them in the next technical grooming.